### PR TITLE
chore: update packages and lint

### PR DESCRIPTION
**Pull request by jwbrandon**
### This PR will...
We have various dependencies that are out of date or unused. This pull request tries to remedy some/most of that.

#### Addresses Issue(s):

JW8-###

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{
+module.exports = {
```
Comment (jwbrandon) [inline .eslintrc.js:1]: Made this a js config so that we can use it in `.eslintrc-tsc.js`

File: .eslintrc-tsc.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const config = require('./.eslintrc.js');
```
Comment (jwbrandon) [inline .eslintrc-tsc.js:1]: We need a separate config for typescript linting as of eslint 8, as eslint now parser typescript but often does it poorly.

File: .eslintrc.js [hunk 3]
```diff
@@ -195,11 +195,16 @@
                 }],
                 "@typescript-eslint/no-explicit-any": 0,
                 "@typescript-eslint/adjacent-overload-signatures": "error",
-                "@typescript-eslint/ban-ts-ignore": "error",
-                "@typescript-eslint/ban-types": "error",
-                "@typescript-eslint/class-name-casing": "error",
+                "@typescript-eslint/ban-ts-comment": "error",
```
Comment (jwbrandon) [inline .eslintrc.js:198]: ban-ts-ignore was renamed to ban-ts-comment

File: .eslintrc.js [hunk 4]
```diff
@@ -195,11 +195,16 @@
                 }],
                 "@typescript-eslint/no-explicit-any": 0,
                 "@typescript-eslint/adjacent-overload-signatures": "error",
-                "@typescript-eslint/ban-ts-ignore": "error",
-                "@typescript-eslint/ban-types": "error",
```
Comment (jwbrandon) [inline .eslintrc.js:199]: ban-types was changed entirely from the originally rule, most of our code was failing with it so it seemed prudent to turn it off.

File: .eslintrc.js [hunk 5]
```diff
@@ -195,11 +195,16 @@
...
-                "@typescript-eslint/class-name-casing": "error",
```
Comment (jwbrandon) [inline .eslintrc.js:200]: this became `naming-convention -> class`

File: .eslintrc.js [hunk 6]
```diff
@@ -195,11 +195,16 @@
...
+                "@typescript-eslint/naming-convention": ["error", {
+                    "selector": "class",
+                    "format": ["PascalCase"]
+                }, {
+                    "selector": "interface",
+                    "format": ["PascalCase"],
+                    "custom": { "regex": "^I[A-Z]", "match": false }
+                }],
                 "@typescript-eslint/consistent-type-assertions": "error",
-                "@typescript-eslint/interface-name-prefix": "error",
```
Comment (jwbrandon) [inline .eslintrc.js:202]: this became `naming-convention -> interface`

File: .npmrc [hunk 7]
```diff
@@ -0,0 +1 @@
+save-exact=true
```
Comment (jwbrandon) [inline .npmrc:1]: We have already agreed that we should do this, this just forces it in npm install without thinking about it.

File: dangerfile.js [hunk 8]
```diff
@@ -1,40 +0,0 @@
-import { danger, error, message, schedule, warn } from 'danger';
```
Comment (jwbrandon) [inline dangerfile.js:1]: `danger` didn't appear to be used anywhere that I could find and the package that we are using is 8 major version out of date with various security issues.

File: package.json [hunk 9]
```diff
@@ -13,59 +13,57 @@
...
-    "karma-chrome-launcher": "2.2.0",
+    "karma-chrome-launcher": "3.1.0",
     "karma-coverage-istanbul-reporter": "2.0.4",
-    "karma-firefox-launcher": "1.1.0",
-    "karma-junit-reporter": "1.2.0",
-    "karma-mocha": "1.3.0",
-    "karma-mocha-reporter": "2.2.5",
-    "karma-safari-launcher": "1.0.0",
+    "karma-firefox-launcher": "2.1.1",
+    "karma-junit-reporter": "2.0.1",
```
Comment (jwbrandon) [inline package.json:39]: All versions updated to match those versions in our internal project.

File: src/js/api/config.js [hunk 10]
```diff
@@ -87,6 +86,7 @@ const Config = function(options, persisted) {
         config.base = getScriptPath('jwplayer.js');
     }
     config.base = (config.base || loadFrom()).replace(/\/?$/, '/');
+    // eslint-disable-next-line
     __webpack_public_path__ = config.base;
```
Comment (jwbrandon) [inline src/js/api/config.js:90]: eslint didn't like use modifying a "global" here, but really we replace this at build time.

File: src/js/jwplayer.js [hunk 11]
```diff
@@ -7,8 +7,8 @@ import { version } from './version';
 import Api from 'api/api';
 import ApiSettings from 'api/api-settings';
 
-/* global __webpack_public_path__:true*/
 /* eslint camelcase: 0 */
+// eslint-disable-next-line
 __webpack_public_path__ = loadFrom();
```
Comment (jwbrandon) [inline src/js/jwplayer.js:12]: eslint didn't like use modifying a "global" here, but really we replace this at build time.

File: src/js/model/simplemodel.ts [hunk 12]
```diff
@@ -22,7 +22,9 @@ export default class SimpleModel extends Events {
     add(attr: string, value: any): void {
         Object.defineProperty(this, attr, {
             get: () => this.attributes[attr],
-            set: (val) => this.set(attr, val),
+            set: (val) => {
```
Comment (jwbrandon) [inline src/js/model/simplemodel.ts:25]: setters shouldn't return a value.

File: src/js/polyfills/webvtt.js [hunk 13]
```diff
@@ -383,7 +383,7 @@ function StyleBox() {
 StyleBox.prototype.applyStyles = function (styles, div) {
     div = div || this.div;
     for (const prop in styles) {
-        if (styles.hasOwnProperty(prop)) {
+        if (Object.prototype.hasOwnProperty.call(styles, prop)) {
```
Comment (jwbrandon) [inline src/js/polyfills/webvtt.js:386]: https://eslint.org/docs/rules/no-prototype-builtins

File: src/js/providers/utils/id3Parser.js [hunk 14]
```diff
@@ -153,7 +153,7 @@ export function parseID3(activeCues = []) {
         }
 
         // These friendly names mapping provides compatibility with our implementation prior to 7.3
-        if (friendlyNames.hasOwnProperty(cue.value.key)) {
+        if (Object.prototype.hasOwnProperty.call(friendlyNames, cue.value.key)) {
```
Comment (jwbrandon) [inline src/js/providers/utils/id3Parser.js:156]: https://eslint.org/docs/rules/no-prototype-builtins

File: package.json [hunk 15]
```diff
@@ -86,7 +84,8 @@
     "watch:css": "npm run build:css -- --watch",
     "watch:tests": "karma start karma.conf.js --no-single-run --auto-watch --no-browsers",
     "lint": "npm-run-all -c -p lint:*",
-    "lint:js": "eslint './src/js' './test/*.js' --ext .ts --ext .js",
+    "lint:js": "eslint './src/js' './test/*.js' --ext .js",
+    "lint:ts": "eslint --no-eslintrc -c .eslintrc-tsc.js './src/js' --ext .ts",
```
Comment (jwbrandon) [inline package.json:88]: New typescript only linting.

File: src/js/utils/underscore.d.ts [hunk 16]
```diff
@@ -12,6 +12,9 @@
 
 /* eslint spaced-comment: 0 */
 /* eslint @typescript-eslint/no-empty-interface: 0 */
+/* eslint @typescript-eslint/no-shadow: 0 */
```
Comment (jwbrandon) [inline src/js/utils/underscore.d.ts:15]: We didn't even write this file. Ideally we would switch to `@types/underscore`, but I tried to do that and it causes a ton of our type definitions to fail. I will add it to our technical debt tracking internally.

File: src/js/parsers/captions/vttparser.js [hunk 17]
```diff
@@ -471,6 +472,7 @@ VTTParser.prototype = {
         } catch (e) {
             throw e;
         }
+        /* eslint-enable no-useless-catch */
```
Comment (imbcmdth) [inline src/js/parsers/captions/vttparser.js:475]: To be fair, why bother catching? 🤣

**Review by imbcmdth — APPROVED** [review]

File: src/js/parsers/captions/vttparser.js (hunk 17 above)
Comment (jwbrandon) [reply src/js/parsers/captions/vttparser.js:475]: I honestly didn't want to change the code here as we might end up pulling this file in from `vtt.js` at some point. Might as well fix it though.

**Pull request merged**